.github/workflows/code-analysis.yml

```diff
@@ -0,0 +1,43 @@
+name: "🔎 Code Scanning"
+# https://github.com/github/codeql-action
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+   branches: [master]
+  # workflow_dispatch:
+  schedule:
+    #        ┌───────────── minute (0 - 59)
+    #        │  ┌───────────── hour (0 - 23)
+    #        │  │ ┌───────────── day of the month (1 - 31)
+    #        │  │ │ ┌───────────── month (1 - 12 or JAN-DEC)
+    #        │  │ │ │ ┌───────────── day of the week (0 - 6 or SUN-SAT)
+    #        │  │ │ │ │
+    #        │  │ │ │ │
+    #        │  │ │ │ │
+    #        *  * * * *
+    - cron: '0 12 * * 1'
+
+jobs:
+  CodeQL:
+    # CodeQL runs on ubuntu-latest, windows-latest, and macos-latest
+    runs-on: ubuntu-latest
+
+    permissions:
+      # required for all workflows
+      security-events: write
+
+      # only required for workflows in private repositories
+      actions: read
+      contents: read
```

Comment:
This is not a private repository, so I think you may drop these two permissions :)